scalameta/contrib/src/main/scala/scala/meta/contrib/instances/EqualityInstances.scala

```diff
@@ -1,26 +1,33 @@
-package scala.meta.contrib.equality
+package scala.meta.contrib.instances
+
+import scala.meta._
+import scala.meta.contrib.Equal
+import scala.meta.prettyprinters.{Show, Structure}
+import scala.language.higherKinds
+
+trait EqualityInstances {
+  implicit def showEquality[F[x] <: Show[x]]: Equal[F] =
+    new Equal[F] {
+      override def isEqual[A](a: A, b: A): Boolean =
+        a.show[F].equals(b.show[F])
```

Comment:
Hmmm... I wonder if we could add a common optimization pattern for show instances. I wonder however if any optimization on string based show instances.

You'll notice Structure is still optimized. This is a just a default implementation as a fallback.